src/components/SingleArticle.jsx

```diff
@@ -1,13 +1,16 @@
 import { useEffect, useState } from 'react';
 import { useParams } from 'react-router-dom';
-import { fetchArticleById, fetchComments } from '../utils/api';
+import { patchArticleVote, fetchArticleById, fetchComments } from '../utils/api';
 import CommentsList from './CommentsList';
 
 export default function SingleArticle() {
   const [ isLoading, setIsLoading ] = useState(true);
   const [ isError, setIsError ] = useState(false);
   const [ article, setArticle ] = useState({});
   const [ comments, setComments ] = useState([]);
+  const [ voteShift, setVoteShift ] = useState(0);
+  const [ hasVoted, setHasVoted ] = useState(false);
```

Comment:
If you are tracking `voteShift` in state this seems like it also accounts for you `hasVoted` state. If voteShift === 0 then they've yet to vote, right? If this handles all of the possible states of your app, which I think is the case,  then you can remove this extra state and base everything on `voteShift`.

This keeps a single source of truth and helps to prevent bugs in the future 👍